src/arc_gahp/io_loop_pthread.cpp

```diff
@@ -574,70 +574,47 @@ IOProcess::stdinPipeHandler()
 Worker*
 IOProcess::createNewWorker(void)
 {
-	Worker *new_worker = NULL;
-	new_worker = new Worker(newWorkerId());
+	int new_id = newWorkerId();
+	auto [it, success] = m_workers_list.insert({new_id, Worker(new_id)});
```

Comment:
I think here is a good place to use `emplace` rather than `insert`, as it will construct the pair in place, rather than construct on the stack + move.